.github/actions/repo-metadata/action.yaml

```diff
@@ -0,0 +1,49 @@
+name: repo-metadata
+description: Retrieve repository metadata using the GitHub API
+inputs:
+  gh-token:
+    description: GitHub token to query the API
+    required: true
+outputs:
+  fork:
+    description: |
+      Repository fork status (true if fork, false otherwise)
+    value: ${{ steps.metadata.outputs.fork }}
+  private:
+    description: |
+      Repository privacy status (true if private, false otherwise)
+    value: ${{ steps.metadata.outputs.private }}
+  default-branch:
+    description: |
+      Default branch of the repository (branch name only, e.g., master)
+    value: ${{ steps.metadata.outputs.default-branch }}
+  archived:
+    description: |
+      Repository archived status (true if archived, false otherwise)
+    value: ${{ steps.metadata.outputs.archived }}
+runs:
+  using: composite
+  steps:
+    - name: Retrieve repository metadata
+      id: metadata
+      shell: bash
+      run: |
+        set -euo pipefail
+        # Query the GitHub API for repository data
+        repo_json=$(
+          curl -sL --fail \
+          -H 'Authorization: Bearer ${{ inputs.gh-token }}' \
+          https://api.github.com/repos/${{ github.repository }}
```

Comment:
preferably replace prefix w/ `${{ github.api_url }}` (that way, this action should also for on GHE)